### PR TITLE
Added Dispose/Disposed to command and event protocol to clean up k3po driver

### PR DESCRIPTION
**Pull request by dpwspoon**
Resolves https://github.com/k3po/k3po/issues/239

### CODE REVIEW

**Comment by dpwspoon** [comment]
Tests are still failing due to other known issues.  I think this is a good product increment.  Work is still needed on tests and other 3.x issues.  Will resolve in other PR.  Looking at k3po.js now.

File: driver/src/main/java/org/kaazing/k3po/driver/internal/Robot.java [hunk 1]
```diff
@@ -218,32 +200,41 @@ public String getObservedScript() {
...
+                        @Override
+                        public void operationComplete(ChannelGroupFuture future) throws Exception {
+                            clientChannels.close();
+                            try {
+                                bootstrapFactory.shutdown();
+                                bootstrapFactory.releaseExternalResources();
+                            } catch (Exception e) {
+                                if (LOGGER.isDebugEnabled()) {
+                                    LOGGER.error("Caught exception releasing resources", e);
+                                }
```
Comment (jfallows) [inline driver/src/main/java/org/kaazing/k3po/driver/internal/Robot.java]: Do you think the `disposedFuture` should fail here? What would that imply for the caller?

File: driver/src/main/java/org/kaazing/k3po/driver/internal/control/handler/ControlEncoder.java [hunk 2]
```diff
@@ -45,7 +49,11 @@ protected Object encode(ChannelHandlerContext ctx, Channel channel, Object messa
         if (message instanceof ControlMessage) {
             ControlMessage controlMessage = (ControlMessage) message;
 
-            switch (controlMessage.getKind()) {
+            Kind kind = controlMessage.getKind();
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Encoding " + kind);
+            }
```
Comment (jfallows) [inline driver/src/main/java/org/kaazing/k3po/driver/internal/control/handler/ControlEncoder.java]: Is this logging still necessary now that you have k3po control protocol tests?

File: driver/src/test/java/org/kaazing/k3po/driver/internal/RobotIT.java [hunk 3]
```diff
@@ -207,7 +205,6 @@ public void shouldAbortConnectOK() throws Exception {
         Pattern p = Pattern.compile(expected);
         assertTrue(p.matcher(observedScript).matches());
```
Comment (jfallows) [inline driver/src/test/java/org/kaazing/k3po/driver/internal/RobotIT.java]: Consider using this approach instead to improve readability.
http://stackoverflow.com/questions/8505153/assert-regex-matches-in-junit

**Comment by jfallows** [comment]
Perhaps bump the control protocol version to `3.0` due to `DISPOSE` and `DISPOSED` ?

**Pull request merged**